File: Java/problems/Power.java
package Java.problems;

public class Power {
    public static void main(String[] args) {
        int result = power(5,2);
        if(result == -1){
            System.out.println("Enter a positive exponent");
        }
        System.out.println(result);
    }

    public static int power(int base, int exponent) {
        if(exponent < 0) return -1;
        if(exponent == 0) return 1;

        return base * power(base, exponent-1);
    }
}

File: Javascript/problems/power.js
//RECURSION

const power = (base, exponent) => {
  if (exponent === 0) return 1
  if (exponent < 0) return 'Enter a positive exponent'

  return base * power(base, exponent - 1)
}

console.log(power(2, 0)) //1
console.log(power(-2, 2)) //4
